assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,320 @@
+  }, [value, options]);
+
+  DropDown.handleClickOutside = () => setOpenOptions(false);
+
+  const clearOptionsRefs = () => {
+    DropDown.arrayOfOptionsRefs = [];
+  };
+
+  const handleOpenOptions = (event) => {
+    switch (event.type) {
+      case 'click':
+        setOpenOptions(!openOptions);
+        setFocusedOption(document.activeElement.id);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          _handleOpenOptions(event);
+        }
+        break;
+      default:
+    }
+  };
+
+  const _handleOpenOptions = useCallback(() => {
+    setOpenOptions(!openOptions);
+  }, [openOptions]);
+
+  const handleCurrentValue = useCallback(
+    (option) => {
+      if (onChange) {
+        onChange(option);
+      }
+    },
+    [onChange]
+  );
+
+  const handleSelectElement = (event, index) => {
+    event.preventDefault();
+    if (index >= 0 && index < DropDown.arrayOfOptionsRefs.length) {
+      DropDown.arrayOfOptionsRefs[index].focus();
+      setFocusedOption(document.activeElement.id);
+    }
+  };
+
+  const handleOptionsEvents = (option, index, event) => {
+    switch (event.type) {
+      case 'click':
+        handleCurrentValue(option);
+        setOpenOptions(!openOptions);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          handleCurrentValue(option);
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else if (event.key === 'ArrowUp') {
+          handleSelectElement(event, index - 1);
+        } else if (event.key === 'ArrowDown') {
+          handleSelectElement(event, index + 1);
+        } else if (event.key === 'Escape') {
+          event.preventDefault();
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else {
+          const searchTerm = searchValue + String.fromCharCode(event.keyCode);
+          setSearchValue(searchTerm);
+          clearSearchValue();
+        }
+        break;
+      default:
+    }
+  };
+
+  const clearSearchValue = useCallback(
+    debounce(800, () => {
+      setSearchValue('');
+    })
+  );
+
+  useEffect(() => {
+    if (searchValue !== '') {
+      const searchIndex = options.findIndex((item) =>
+        item.name.toLowerCase().includes(searchValue.toLowerCase())
+      );
+      if (searchIndex >= 0) {
+        DropDown.arrayOfOptionsRefs[searchIndex].focus();
+      }
+    }
+  }, [searchValue, options]);
+
+  useEffect(() => {
+    if (openOptions) {
+      DropDown.arrayOfOptionsRefs[0].focus();
+    }
+  }, [openOptions]);
+
+  const setOptionRef = (element) => {
+    if (element !== null) {
+      DropDown.arrayOfOptionsRefs.push(element);
+    }
+  };
+
+  const setsize = options.length;
+
+  return (
+    <DropDownContainer>
+      <DropDownSelect
+        onClick={handleOpenOptions}
+        onKeyDown={handleOpenOptions}
+        aria-pressed={openOptions}
```

Comment:
I think we need to follow the example in [listbox_role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role).

Reply:
Some aria informations added.